File: test/interface/golden/sum/OnOrOff.ml
type onOrOff =
  | On
  | Off

let encodeOnOrOff x =
  match x with
  | On ->
     Aeson.Encode.string "On"
  | Off ->
     Aeson.Encode.string "Off"

let decodeOnOrOff json =
  match Js_json.decodeString json with
  | Some "On" -> Js_result.Ok On
  | Some "Off" -> Js_result.Ok Off
  | Some err -> Js_result.Error ("decodeOnOrOff: unknown enumeration '" ^ err ^ "'.")
  | None -> Js_result.Error "decodeOnOrOff: expected a top-level JSON string."

File: test/interface/golden/lib/js/sum/OnOrOff.js
'use strict';

var Block   = require("bs-platform/lib/js/block.js");
var Js_json = require("bs-platform/lib/js/js_json.js");

function encodeOnOrOff(x) {
  if (x !== 0) {
    return "Off";
  } else {
    return "On";
  }
}

function decodeOnOrOff(json) {
  var match = Js_json.decodeString(json);
  if (match) {
    var err = match[0];
    switch (err) {
      case "Off" : 
          return /* Ok */Block.__(0, [/* Off */1]);
      case "On" : 
          return /* Ok */Block.__(0, [/* On */0]);
      default:
        return /* Error */Block.__(1, ["decodeOnOrOff: unknown enumeration '" + (err + "'.")]);
    }
  } else {
    return /* Error */Block.__(1, ["decodeOnOrOff: expected a top-level JSON string."]);
  }
}

exports.encodeOnOrOff = encodeOnOrOff;
exports.decodeOnOrOff = decodeOnOrOff;
/* No side effect */
